packages/frontend/src/components/UserSettings/AppearancePanel/index.tsx

```diff
@@ -1,29 +1,58 @@
-import { Intent, Spinner } from '@blueprintjs/core';
+import { Spinner } from '@blueprintjs/core';
 import { subject } from '@casl/ability';
 import { ECHARTS_DEFAULT_COLORS } from '@lightdash/common';
-import { FC, useCallback, useEffect, useState } from 'react';
-
+import {
+    Box,
+    Button,
+    ColorInput,
+    SimpleGrid,
+    Stack,
+    Title,
+} from '@mantine/core';
+import { useForm } from '@mantine/form';
+import { FC, useEffect } from 'react';
 import { useOrganization } from '../../../hooks/organization/useOrganization';
 import { useOrganizationUpdateMutation } from '../../../hooks/organization/useOrganizationUpdateMutation';
-import { InputWrapper } from '../../ChartConfigPanel/ChartConfigPanel.styles';
 import { Can, useAbilityContext } from '../../common/Authorization';
-import ColorInput from '../../common/ColorInput';
-import {
-    AppearancePanelWrapper,
-    ColorPalette,
-    SaveButton,
-    Title,
-} from './AppearancePanel.styles';
 
 const AppearancePanel: FC = () => {
     const ability = useAbilityContext();
     const { isLoading: isOrgLoading, data } = useOrganization();
     const updateMutation = useOrganizationUpdateMutation();
-    let [colors, setColors] = useState<string[]>(
-        data?.chartColors || ECHARTS_DEFAULT_COLORS.slice(0, 8),
-    );
+    const IS_HEX_CODE_COLOR_REGEX = /^#([a-fA-F0-9]{3}|[a-fA-F0-9]{6})$/;
+
+    const getColorFields = (colors: string[]) =>
+        colors.reduce(
+            (acc, color, index) => ({ ...acc, [`color${index}`]: color }),
+            {},
+        );
```

Comment:
generates `color0` to `color7` fields, unless we want to increment the `index` by 1 to match what the user sees. what do we think? 🤔

Reply:
match what the user sees is better imo

Reply:
im not entirely sure why we need this

Reply:
(also can be outside of the component function)

Reply:
done @yasmin-e , thanks!

> im not entirely sure why we need this

@IrakliJani, I want to set the name of the fields for the form, so it looks like this: 
```
{
color1: colors[0]
color2:  colors[1]
... 
color8: colors[7]
}
```

how would you do this? I thought about just writing it (no loops or wtv) but not sure what to do here or if I'm just missing something 😞